File: app/src/main/java/com/arnaudpiroelle/manga/provider/mangapanda/MangaPandaModule.java
package com.arnaudpiroelle.manga.provider.mangapanda;

import com.arnaudpiroelle.manga.provider.mangapanda.api.MangaPandaApiService;
import com.arnaudpiroelle.manga.provider.mangapanda.api.MangaPandaDataApiService;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import retrofit.RestAdapter;
import retrofit.client.OkClient;

import static com.arnaudpiroelle.manga.BuildConfig.MANGAPANDA_BASE_URL;
import static com.arnaudpiroelle.manga.BuildConfig.MANGAPANDA_CDN_BASE_URL;
import static retrofit.RestAdapter.LogLevel.NONE;

@Module
public class MangaPandaModule {

    @Provides
    @Singleton
    public MangaPandaApiService providesMangaPandaApiService(OkClient client) {
        RestAdapter restAdapter = new RestAdapter.Builder()
                .setEndpoint(MANGAPANDA_BASE_URL)
                .setLogLevel(NONE)
                .setClient(client)
                .build();

        return restAdapter.create(MangaPandaApiService.class);
    }

    @Provides
    @Singleton
    public MangaPandaDataApiService providesMangaPandaDataApiService(OkClient client) {
        RestAdapter restAdapter = new RestAdapter.Builder()
                .setEndpoint(MANGAPANDA_CDN_BASE_URL)
                .setLogLevel(NONE)
                .setClient(client)
                .build();

        return restAdapter.create(MangaPandaDataApiService.class);
    }
}

File: app/src/main/kotlin/com/arnaudpiroelle/manga/provider/mangapanda/MangaPandaModule.kt
package com.arnaudpiroelle.manga.provider.mangapanda

import com.arnaudpiroelle.manga.provider.mangapanda.api.MangaPandaApiService
import com.arnaudpiroelle.manga.provider.mangapanda.api.MangaPandaDataApiService

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import retrofit.RestAdapter
import retrofit.client.OkClient

import com.arnaudpiroelle.manga.BuildConfig.MANGAPANDA_BASE_URL
import com.arnaudpiroelle.manga.BuildConfig.MANGAPANDA_CDN_BASE_URL
import retrofit.RestAdapter.LogLevel.NONE

@Module
class MangaPandaModule {

    @Provides
    @Singleton
    fun providesMangaPandaApiService(client: OkClient): MangaPandaApiService {
        val restAdapter = RestAdapter.Builder().setEndpoint(MANGAPANDA_BASE_URL).setLogLevel(NONE).setClient(client).build()

        return restAdapter.create(MangaPandaApiService::class.java)
    }

    @Provides
    @Singleton
    fun providesMangaPandaDataApiService(client: OkClient): MangaPandaDataApiService {
        val restAdapter = RestAdapter.Builder().setEndpoint(MANGAPANDA_CDN_BASE_URL).setLogLevel(NONE).setClient(client).build()

        return restAdapter.create(MangaPandaDataApiService::class.java)
    }
}
